app/controllers/TaxFreeAmountController.scala

```diff
@@ -50,17 +53,21 @@ class TaxFreeAmountController @Inject()(codingComponentService: CodingComponentS
   def taxFreeAmount: Action[AnyContent] = (authenticate andThen validatePerson).async {
     implicit request =>
       val nino = request.taiUser.nino
-
+      val totalTaxFuture = taxAccountService.totalTax(nino, TaxYear())
       (for {
         codingComponents <- codingComponentService.taxFreeAmountComponents(nino, TaxYear())
         employmentNames <- employmentService.employmentNames(nino, TaxYear())
         companyCarBenefits <- companyCarService.companyCarOnCodingComponents(nino, codingComponents)
+        totalTax <- totalTaxFuture
       } yield {
-        val viewModel = TaxFreeAmountViewModel(codingComponents, employmentNames, companyCarBenefits)
+        totalTax match {
+          case TaiSuccessResponseWithPayload(totalTax: TotalTax) =>
+        val viewModel = TaxFreeAmountViewModel(codingComponents, employmentNames, companyCarBenefits, totalTax)
```

Comment:
Need to handle the failure case here, the recover won't catch it

Reply:
Done